File: cpp/p6.cpp
#include "euler.h"

using namespace std;

int main() {
  int sum = 0;
  int sum_of_squares = 0;
	
  for( int i = 1; i <= 100; i++ ) {
    sum += i;
    sum_of_squares += i * i;
  }
	
  cout << ( sum * sum ) - sum_of_squares << endl;
}

File: clojure/p6.clj
(ns p6
  (:use [euler :only (naturals sqr sum timed-test)]))

(timed-test
 "Problem 6"
 25164150
 (let [nums (take 100 (naturals))
       sum-of-squares (sum (map sqr nums))
       square-of-sum (sqr (sum nums))]
   (- square-of-sum sum-of-squares)))
